### DISCUSSION TITLE
Server actions bind data to form not working

### DISCUSSION THREAD

**Discussion by FlyingTurkman** [vercel/next.js#56904] [Help]
### Summary

I am trying to basic messenger with server actions but I can't bind a data to form data

Message sender function:

```tsx
'use server'


export async function sendMessage(formData: FormData, userId?: string) {
    console.log(formData, userId)
}
```

component:
```tsx
const sendMessageWithUserId = sendMessage.bind(null, userId)
```

On this line giving Type 'undefined' or 'string' cannot be assigned to type 'FormData'. error

**Comment by Xu-Justin** [comment]
Did you find the answer? I encounter same error.

**Reply by FlyingTurkman** [reply]
Yeah i am not using bind data.

**Reply by darvid** [reply]
for anyone who lands here, evidently `formData` needs to be the **last param** in the server action

**Reply by Xu-Justin** [reply]
Any references or docs related to that?

**Reply by darvid** [reply]
Not that I could find.
